operator/src/test/java/org/bf2/operator/managers/StrimziManagerTest.java

```diff
@@ -205,6 +211,30 @@ public void testUninstallingStrimziOperator() {
         assertFalse(checkStrimziVersion(strimziVersions, "strimzi-cluster-operator.v2", true));
     }
 
+    @Test
+    public void testStrimziLeadershipLeaseCleanedOnDelete() {
+        final Lease lease = new LeaseBuilder()
+                .withNewMetadata()
+                    .withName("strimzi-cluster-operator.v1-leadership-token")
+                    .withNamespace("ns-1")
+                .endMetadata()
+                .withNewSpec()
+                    .withHolderIdentity("me")
+                    .withLeaseDurationSeconds(1000)
+                    .withAcquireTime(ZonedDateTime.ofInstant(Instant.now(), ZoneId.of("UTC")))
+                .endSpec().build();
+        server.getClient().leases().inNamespace("ns-1").create(lease);
+        installStrimziOperator("strimzi-cluster-operator.v1", "ns-1", "2.7.0=kafka-2.7.0", true, true);
+        Lease actual = server.getClient().leases().withName("strimzi-cluster-operator.v1-leadership-token").get();
+        assertNull(actual);
```

Comment:
realized the test is incorrect here, also I need to figure out how to trigger the onDelete watcher since the uninstallStrimziOperator doesn't trigger it as hoped